aviary/subsystems/propulsion/turboprop_model.py

```diff
@@ -286,37 +337,256 @@ def setup(self):
             promotes_outputs=[('turboprop_thrust', Dynamic.Mission.THRUST)],
         )
 
-        max_thrust_group.add_subsystem(
+        self.add_subsystem(
             'max_thrust_adder',
             subsys=max_thrust_adder,
             promotes_inputs=['*'],
-            promotes_outputs=[('turboprop_thrust_max', Dynamic.Mission.THRUST_MAX)]
-        )
-
-        self.add_subsystem(
-            'turboprop_max_group',
-            max_thrust_group,
-            promotes_inputs=['*'],
-            promotes_outputs=[Dynamic.Mission.THRUST_MAX],
+            promotes_outputs=[('turboprop_thrust_max', Dynamic.Mission.THRUST_MAX)],
         )
 
     def configure(self):
-        # configure step to alias thrust output from shaft power model if present
+        """
+        Correctly connect variables between shaft power model, gearbox, and propeller,
+        aliasing names if they are present in both sets of connections.
+
+        If a gearbox is present, inputs to the gearbox are usually done via connection,
```

Comment:
So essentially without gearbox: Motor -> promoted -> Prop
with gearbox: Motor -> connection -> Gearbox -> promotion -> Prop
?

Reply:
my preference would be:
motor/engine deck/engine model output: `power_source_rpm`
gearbox input: `power_source_rpm`
gearbox output: `prop_rpm`
propeller input: `prop_rpm`

If the gearbox exists, all connections are made automatically by promotions
If there is no gearbox, then a connection or alias is used.

I think that will be significantly more clear to users and could also simplify the code here